format.go

```diff
@@ -234,6 +235,25 @@ func isDuration(v interface{}) bool {
 	return ok && len(s) == 0 && len(units) > 0 && strings.Index("HMS", units) != -1
 }
 
+// isPeriod tells whether given string is a valid period format
+// from the ISO 8601 ABNF as given in Appendix A of RFC 3339.
+//
+// see https://datatracker.ietf.org/doc/html/rfc3339#appendix-A, for details
+func isPeriod(v interface{}) bool {
+	s, ok := v.(string)
+	if !ok {
+		return true
+	}
+	ss := strings.SplitN(s, "/", 3)
```

Comment:
since it is expected to have only 2 splits, use indexOf and slicing (just to avoid confusion)
name ss[0] to start and ss[1] to end so that they match the naming in RFC.

Reply:
Problem with matching the RFC naming is that the parts don't have a stable name in it; either part can be `iso-date-time` or `duration`. I opted for use `start` and `end` to match their semantic meaning (to some extent).

Reply:
`start` and `end` are fine